---
title: "Untangling molecular food-webs to investigate the impacts of _Dikerogammarus villosus_ in the UK"
author: "Marco Benucci"
date: "3 September 2019"
output: pdf_document
---

```{r knitr setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r environment setup, include=F}
## R(v3.5.1 - Win)
#.libPaths(c("D:\\Dropbox\\PhD Hull\\PhD docs\\Thesis\\R_stats\\rlib3.5_win", .libPaths("D:\\Dropbox\\PhD Hull\\PhD docs\\Thesis\\R_stats\\rlib3.5_win")))

## R(v3.5.1 - OSX)
#.libPaths(c("D:\\Dropbox\\PhD Hull\\PhD docs\\Thesis\\R_stats\\rlib3.5_osx", .libPaths("D:\\Dropbox\\PhD Hull\\PhD docs\\Thesis\\R_stats\\rlib3.5_osx")))

## R(v3.5.1 - Ubuntu)
.libPaths(c("/home/mb/Dropbox/PhD Hull/PhD docs/Thesis/R_stats/rlib3.5_unix", .libPaths("/home/mb/Dropbox/PhD Hull/PhD docs/Thesis/R_stats/rlib3.5_unix")))
```

```{r directory setup, include=FALSE}
dir()
rm(list=ls())
ls()
```

#Setting working directory and packages

The analysis was done on R (R Core Team, 2018) using the following environment and packages.

```{r packages loading, include=F}
pack.list = c("aod","ape","bipartite","cowplot","devtools","dplyr","EcoSimR","ggplot2","ggsignif","gplots","grid","gridExtra","iNEXT","lattice","lme4","lmtest","MASS","mgcv","msa","network","plyr","reshape2","spaa","stringi","tidyr","UpSetR","vegan","zoo")
new.packages = pack.list[!(pack.list %in% installed.packages()[,"Package"])]
if(length(new.packages)) install.packages(new.packages, repos="https://cran.rstudio.com")
lapply(pack.list, require, character.only=T)

#"AED","igraph","msa","phyloseq","ggnetwork"
```

```{r session info}
sessionInfo()
```

# 1. Input files and Quality Control.

The samples were processed on two different sequencing runs. 28 February corresponds to the feeding trials involving 2 of the 3 prey species; 4 August instead include the last prey species plus the wild samples.

```{r input files}
raw_feb = read.csv("data/CO1-Dvfeb-merge-forwonly_nonchimera-c1-cov2-id97-by-taxonomy-readcounts.blast.csv")
raw_aug = read.csv("data/CO1Aug17-merge-forwonly-nonchimera-c97-cov2_id97-by-taxonomy-readcounts.blast.csv")
head(raw_feb)
head(raw_aug)
```

```{r taxonomy file, include=T}
## Custom function to format the taxonomy column

format_taxonomy <- function(input_data){
  for (t in as.data.frame(input_data)){
    n = gsub("\\['|\\']", "", t)
    i = gsub("\\'", "", n)
    taxa.split = data.frame(cbind(i))
  }
  return(taxa.split)
}

## First input file
taxonomy.feb = raw_feb$taxomomy
raw_feb = raw_feb[,!(colnames(raw_feb) %in% 'taxomomy')]

taxa.feb=format_taxonomy(taxonomy.feb)
taxa.feb=data.frame(taxa.feb)

## Second input file
taxonomy.aug = raw_aug$taxomomy
raw_aug = raw_aug[,!(colnames(raw_aug) %in% 'taxomomy')]

taxa.aug=format_taxonomy(taxonomy.aug)
taxa.aug=data.frame(taxa.aug)

## Writing taxonomy outputs
write.csv(taxa.feb, "R_outputs/CO1DvFeb_taxonomy.csv")
write.csv(taxa.aug, "R_outputs/CO1DvAug_taxonomy.csv")
```

Each initial input file was generated as the result of two separate operations. First DNA centroids were assigned against reference database, the reads that resulted unassigned were then assigned against the NCBI nt database. Each operation generated two separate files which were then merged. The following step is to remove potential duplicate taxa records that would prevent R from definying the row names as the taxa names. Because we removed the last column with the taxonomy information, we can use the whole frame; otherwise we would have needed to avoid the last column in each frame.

```{r removing duplicates, include=T}
raw_feb = as.data.frame(raw_feb %>% group_by(X.OTU.ID) %>% summarise_all(list(sum)))
raw_aug = as.data.frame(raw_aug %>% group_by(X.OTU.ID) %>% summarise_all(list(sum)))
```

We can now set up the rownames for each data frame equal to the species names, and we can remove the first column.

```{r data frames rownames, include=F}
## Setting up rownames equal to the OTU ID
rownames(raw_feb) = raw_feb$X.OTU.ID
rownames(raw_aug) = raw_aug$X.OTU.ID

## Removing the OTU ID column
dv_feb = raw_feb[,!(colnames(raw_feb) %in% 'X.OTU.ID')]
dv_aug = raw_aug[,!(colnames(raw_aug)%in% 'X.OTU.ID')]
```

# 2. Read depth per sample

Since the data were generated into three separate sequencing runs, we check the read depth of all the samples across the three separate frames. We first remove all empty samples before calculating mean and standard deviation values across all three separate frames. Total number of samples were respectively, `dv_feb` (sequencing run from February 2017) Ntot=320; `dv_aug` (sequencing run from Augusut 2017) Ntot=350.

```{r samples, include=T}
## First frame
feb_reads = dv_feb

## Second frame
aug_reads = dv_aug
```


```{r intial read depth boxplot}
feb_melt = melt(t(feb_reads))
aug_melt = melt(t(aug_reads))

feb_melt$library = "First"
aug_melt$library = "Second"

depth_melt = full_join(feb_melt, aug_melt)

depth_melt = as.data.frame(depth_melt[,!c(colnames(depth_melt) %in% "Var2")] %>% group_by(Var1, library) %>% summarise_all(list(sum)))
```

```{r read depth boxplot}
ggplot(depth_melt, aes(library, value)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(alpha=0.4)
```

```{r mean read depth, include=F}
#feb_reads = subset(feb_reads, select=colSums(feb_reads) > 0)
#aug_reads = subset(aug_reads, select=colSums(aug_reads) > 0)

## Mean values
mean(colSums(feb_reads))
mean(colSums(aug_reads))

hist(colSums(feb_reads), breaks=20)
hist(colSums(aug_reads), breaks=20)
```

Considering our normalisation steps during the libraries preparations, we expect the read coverage per sample to be normally distributed in each run. With this in mind we tested for normality and then we tested for differences in the read depth between the different plates. 

The results show that all the plates were not significantly difference in regards to reads depth: `ha.reads` and `ha1.reads` (t-test: t = 0.62067, df = 142, p-value = 0.5358), `ha.reads` and `ha2.reads` (t-test: t = -0.95017, df = 407, p-value = 0.3426), `ha1.reads` and `ha2.reads` (t-test: t = -1.7813, df = 439, p-value = 0.07555).

```{r reads distribution, include=T, echo=F}
shapiro.test(colSums(feb_reads))
shapiro.test(colSums(aug_reads))

t.test(colSums(feb_reads), colSums(aug_reads), var.equal = T)
```

To remove low coverage samples we decide to retain samples that contained N reads greater than mean-sd. This was possible for only the data from the run done in February 2017. For the data frame from August 2017, the mean-sd value came back as negative, so we decided to apply an arbitrary fixed threshold of 2000 reads.

```{r read depth filter, include=T}
cov_feb = mean(colSums(feb_reads))-sd(colSums(feb_reads))
cov_aug = mean(colSums(aug_reads))-sd(colSums(aug_reads))

feb_cover = subset(feb_reads, select = colSums(feb_reads) >= 1000)
aug_cover = subset(aug_reads, select = colSums(aug_reads) >= 100)
```

After this initial low coverage filter we retained 98.5%, and 99.5% of the total number of reads; and respectively 75%, and 75.2% number of samples.

```{r reads survival boxplot, include=T}
feb_cmelt = melt(t(feb_cover))
aug_cmelt = melt(t(aug_cover))

feb_cmelt$library = "First"
aug_cmelt$library = "Second"

cover_melt = full_join(feb_cmelt, aug_cmelt)

cover_melt = as.data.frame(cover_melt[,!c(colnames(cover_melt) %in% "Var2")] %>% group_by(Var1, library) %>% summarise_all(list(sum)))

#boxplot(colSums(feb_cover), colSums(aug_cover), ylab="N reads", xlab="Data frames", main = "Read depth of D.villosus metabarcoding samples greater than mean-sd", names=c("feb_cover\n(Feb2017)", "aug_cover\n(Aug2017)"), las=1, cex.axis=0.8)
```

```{r cover boxplot}
ggplot(cover_melt, aes(library, value)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(alpha=0.4)
```

Read survival is set at 98.8% for the first library and 99.5% for the second.

```{r read survival percentage, include=T}
(sum(feb_cover)/sum(feb_reads))*100
(sum(aug_cover)/sum(aug_reads))*100
```

# 3. Removal of untarget sample and taxa

We check first presence of unwanted taxa and high rank assignment into our data frames. In particular `unassigned` reads, and higher taxa such as unknown `Eukaryota`, `Fungi` and `Bacteria` are going to be removed since they won't be informative at all further downstream. To make the process efficient, we first remove all taxa that have no reads at all as a consequence of previous quality control steps, we then check for the presence of unwanted taxa.

```{r samples, include=T}
## First frame
#colnames(feb_cover)
feb_samples = feb_cover[,-c(grep("WB|GW|RB", colnames(feb_cover)))]

## Second frame
#colnames(aug_cover)
aug_samples = aug_cover[,-c(grep("Ha|CH", colnames(aug_cover)))]
```

Before removing any unwanted taxa we subset both data frame to retain only taxa that have N reads greater than 0. The first data frame includes only feeding trials, therefore we need want to keep all taxa excluding algae, since those were used as food for the _D. villosus_ specimens. We keep potential taxa that might have been lab contaminants, specifically `Philodina_roseola`. The second data frame instead, contains few feeding trials samples plus the gut contents samples of wild `D. villosus` and wild `G. tigrinus`, and the eDNA samples. In this latter frame we then need to expand the list to explude the higher ranks.

```{r removing unwanted taxa}
# First frame
feb_samples = subset(feb_samples, subset = rowSums(feb_samples) > 0)
#rownames(feb_samples)
#rowSums(feb_samples)

discard.feb = c("Thalassiosira_pseudonana","unassigned")

feb_clean = feb_samples[!rownames(feb_samples) %in% discard.feb,]

# Second frame
aug_samples = subset(aug_samples, subset = rowSums(aug_samples) > 0)
#rownames(aug_samples)
#rowSums(aug_samples)

discard.aug = c("Abramis_brama","Anser","Aphanomyces","Arthropoda_environmental_sample","Batrachospermum_gelatinosum","Chromulinales","Cochliopodium","Cochliopodium_actinophorum","Cyclostephanos_dubius","Cyclostephanos_invisitatus","Cyclostephanos_sp._WTC16","Didymella_pinodes","Dinophyceae","Encyonema","Eukaryota","Fistulifera_solaris","Fusarium","Gomphonema_parvulum","Gymnocephalus","Hominidae","Homo_sapiens","Ilyonectria_destructans","invertebrate_environmental_sample","Lagenidium","Leptolegnia_sp._BOLD:AAX5717","Mallomonas_sp.","Melosira_ambiqua","Metazoa","Mus_musculus","Oncorhynchus_mykiss","Oomycetes","Paraphysomonas_sp.","Penicillium_sclerotiorum","Phytophthora","Pinnularia_neomajor","Pseudorasbora_parva","Pythium","Pythium_marsipium","Saprolegnia_delica","Saprolegniaceae","Sellaphora_obesa","Skeletonema_potamos","Sordariomycetes","Stephanodiscus","Stephanodiscus_cf._minutulus","Sus_scrofa","Tetracladium","Tetradesmus_obliquus","Thalassiosira_pseudonana","Urocentrum_turbo","unassigned","Yamagishiella_unicocca","Woloszynskia","zooplankton_environmental_sample")

aug_clean = aug_samples[!rownames(aug_samples) %in% discard.aug,]
```

We retained approx. 88.6%, and 54.1% of the initial N reads.

```{r retained reads and samples, include=T}
(sum(feb_clean)/sum(feb_reads))*100
(sum(aug_clean)/sum(aug_reads))*100
```

# 4. Contamination QC

First data frame contains data from the sequencing run from February 2017, which includes feeding trials of _Dikerogammarus villosus_. To calculate the contamination threshold we decided to account for all laboratory contaminations (e.g. Human DNA) in our `POSITIVE` samples.

```{r threshold first frame, include=T}
## Positive samples first frame
feb_pos = feb_clean[,grep("POS", colnames(feb_clean))]
colSums(feb_pos)

rownames(feb_pos)

feb.pre.filter = as.data.frame(t(feb_clean))
feb.pre.filter = feb.pre.filter[,]/rowSums(feb.pre.filter)
feb.pre.filter[is.na(feb.pre.filter)] <- 0

thres =  0.001
```

We have to exclude the contamination from _Dikerogammarus villosus_ as that would be too high for a fixed contamination threshold and we would lose any informations about prey-predator interactions. The remaining contaminations appear to be: _Asellus aquaticus_ (0.098%), and _Homo sapiens_ (0.13%). Based on the levels of these contaminations we decided to apply a threshold of 0.1% despite this will maintain some levels of contamination from Human DNA.

```{r filter first frame, include=T}
feb.post.filter = as.data.frame(feb.pre.filter)
feb.post.filter[,][feb.post.filter <= thres] <- 0

feb_clean = as.data.frame(t(feb_clean))
feb_clean[,][feb_clean[,]/rowSums(feb_clean) <= thres] <- 0

pn.feb = feb_clean[grep("POS|NEG", rownames(feb_clean)),]
feb_clean = feb_clean[!c(rownames(feb_clean) %in% rownames(pn.feb)),]
feb.pre.filter = feb.pre.filter[!c(rownames(feb.pre.filter) %in% rownames(pn.feb)),]
feb.post.filter = feb.post.filter[!c(rownames(feb.post.filter) %in% rownames(pn.feb)),]
```

Checking for further contaminations in the first dataframe (containing the feeding trials) show no further contamination present.

```{r further contamination}
colSums(feb.post.filter)

## Writing pre- and post-filter data to output
write.csv(feb_clean, "R_outputs/DvFeb_cleaned_reads.csv")
write.csv(feb.pre.filter, "R_outputs/DvFeb_pre-filter.csv")
write.csv(feb.post.filter, "R_outputs/DvFeb_post-filter.csv")
```

In the second data frame the contamination from _Dikerogammarus villosus_ stands at 0.047%, _Gammarus zaddachi_ at 0.05%, and _Homo sapiens_ at 0.02% and 0.01%. Based on the levels of these contaminations we decided to apply a threshold of 0.03% despite this will maintain some levels of contamination from _D. villosus_ DNA. Positive sample 3 failed in amplifying any _Osmia bicornis_ DNA and contains only _D. villosus_ DNA, so we have to exclude it from the QC.

```{r threshold first frame, include=T}
## Positive samples second frame
aug_pos = aug_clean[,grep("POS", colnames(aug_clean))]
colSums(aug_pos)

aug_pos = subset(aug_pos, subset=rowSums(aug_pos) > 0)

aug_pos = aug_pos[,!c(colnames(aug_pos) == "POSITIVE.3")]
aug_pos[,]/colSums(aug_pos)

colSums(aug_pos["Osmia_bicornis",])
colSums(aug_pos)-colSums(aug_pos[c("Osmia_bicornis","Hymenoptera"),])
(colSums(aug_pos)-colSums(aug_pos[c("Osmia_bicornis","Hymenoptera"),]))/colSums(aug_pos)


aug.pre.filter = as.data.frame(t(aug_clean))
aug.pre.filter = aug.pre.filter[,]/rowSums(aug.pre.filter)
aug.pre.filter[is.na(aug.pre.filter)] <- 0

thres = 0.0005
```

We have to exclude the contamination from _Dikerogammarus villosus_ as that would be too high for a fixed contamination threshold and we would lose any informations about prey-predator interactions. The remaining contaminations appear to be: _Asellus aquaticus_ (0.098%), and _Homo sapiens_ (0.13%). Based on the levels of these contaminations we decided to apply a threshold of 0.1% despite this will maintain some levels of contamination from Human DNA.

```{r filter first frame, include=T}
aug.post.filter = as.data.frame(aug.pre.filter)
aug.post.filter[,][aug.post.filter <= thres] <- 0

aug_clean = as.data.frame(t(aug_clean))
aug_clean[][aug_clean[,]/rowSums(aug_clean) <= thres] <- 0

pn.aug = aug_clean[grep("POS|NEG", rownames(aug_clean)),]
aug_clean = aug_clean[!c(rownames(aug_clean) %in% rownames(pn.aug)),]
aug.pre.filter = aug.pre.filter[!c(rownames(aug.pre.filter) %in% rownames(pn.aug)),]
aug.post.filter = aug.post.filter[!c(rownames(aug.post.filter) %in% rownames(pn.aug)),]
```

Visual check and we still have some contamination from PCR, labeled as `cont.17` till `cont.20`. Three of those contain _D.villosus_ DNA only (N=3244, N=38246, N=40021), while one containes _D.villosus_ (N=3304), _Osmia bicornis_ (N=867), and Hymenoptera (N=5). Since _D. villosus_ is the traget predator and _O. bicornis_ being the positive we decide to actually remove the contamination samples without too much impact.

```{r further contamination}
rownames(aug.post.filter)

pcr_cont = as.data.frame(aug_clean[grep("cont", rownames(aug_clean)), colSums(aug_clean[grep("cont", rownames(aug_clean)),]) > 0])

colnames(aug_clean)

aug_clean = as.data.frame(aug_clean[!c(rownames(aug_clean) %in% rownames(pcr_cont)),!c(colnames(aug_clean) %in% c("Harmonia_axyridis","Homo_sapiens","Osmia_bicornis"))])

aug.post.filter = as.data.frame(aug.post.filter[!c(rownames(aug.post.filter) %in% rownames(pcr_cont)),!c(colnames(aug_clean) %in% c("Harmonia_axyridis","Homo_sapiens","Osmia_bicornis"))])

## Writing pre- and post-filter data to output
write.csv(aug_clean, "R_outputs/DvAug_cleaned_reads.csv")
write.csv(aug.pre.filter, "R_outputs/DvAug_pre-filter.csv")
write.csv(aug.post.filter, "R_outputs/DvAug_post-filter.csv")
```

We add now information regarding sampling sites (extracted from the samples names), use of blocking primers or not, pre- or post-filter, and the sample type (gut contents or community)

```{r adding variables first frame, include=F}
feb.pre.filter$type = "pre.filter"
feb.post.filter$type = "post.filter"

feb_clean$sample_id = rownames(feb_clean)
feb.pre.filter$sample_id = rownames(feb_clean)
feb.post.filter$sample_id = rownames(feb_clean)

feb_clean$sample_type = "feeding trial"
feb.pre.filter$sample_type = "feeding trial"
feb.post.filter$sample_type = "feeding trial"
```

We add now information regarding sampling sites (extracted from the samples names), use of blocking primers or not, pre- or post-filter, and the sample type (gut contents or community)

```{r adding variables first frame, include=F}
aug.pre.filter$type = "pre.filter"
aug.post.filter$type = "post.filter"

aug_clean$sample_id = rownames(aug_clean)
aug.pre.filter$sample_id = rownames(aug.pre.filter)
aug.post.filter$sample_id = rownames(aug.post.filter)

aug_clean$sample_type = "NA"
aug_clean[grep("e", aug_clean$sample_id), "sample_type"] = "eDNA"
aug_clean[grep("e", aug_clean$sample_id, invert=T), "sample_type"] = "gut contents"
aug_clean[grep("DvG", aug_clean$sample_id), "sample_type"] = "feeding trial"

aug.pre.filter$sample_type = "NA"
aug.pre.filter[grep("e", aug.pre.filter$sample_id), "sample_type"] = "eDNA"
aug.pre.filter[grep("e", aug.pre.filter$sample_id, invert=T), "sample_type"] = "gut contents"
aug.pre.filter[grep("DvG", aug.pre.filter$sample_id), "sample_type"] = "feeding trial"

aug.post.filter$sample_type = "NA"
aug.post.filter[grep("e", aug.post.filter$sample_id), "sample_type"] = "eDNA"
aug.post.filter[grep("e", aug.post.filter$sample_id, invert=T), "sample_type"] = "gut contents"
aug.post.filter[grep("DvG", aug.post.filter$sample_id), "sample_type"] = "feeding trial"
```

# 5. Feeding trials analysis

We split the feeding trials by prey taxa: _Asellus aquaticus_ (DvA), _Daphnia magna_ (DvD), and _Crangonyx pseudogracilis_ (DvG)

```{r setting up pcr and trial replicates}
dvd = feb_clean[grep("DvD", rownames(feb_clean)),]
dva = feb_clean[grep("DvA", rownames(feb_clean)),]
dvc = aug_clean[grep("DvG", rownames(aug_clean)),]

# Feeding trials with Daphnia

#removing pcr and sample replicate numbers
dvd$sample_id = gsub("[.][0-9]{1}$", "", gsub("[.][0-9]{1}$", "", dvd$sample_id))

#setting pcr and sample repliate numbers in separate columns
dvd$trial_repl = "NA"
dvd[grep("C", rownames(dvd)), "trial_repl"] = "control"
dvd[grep("DvD[0-9]{1,2}[.]1[.]{0,1}[0-9]{0,1}", rownames(dvd)), "trial_repl"] = 1
dvd[grep("DvD[0-9]{1,2}[.]2[.]{0,1}[0-9]{0,1}", rownames(dvd)), "trial_repl"] = 2
dvd[grep("DvD[0-9]{1,2}[.]3[.]{0,1}[0-9]{0,1}", rownames(dvd)), "trial_repl"] = 3
dvd[grep("DvD[0-9]{1,2}[.]4[.]{0,1}[0-9]{0,1}", rownames(dvd)), "trial_repl"] = 4
dvd[grep("DvD[0-9]{1,2}[.]5[.]{0,1}[0-9]{0,1}", rownames(dvd)), "trial_repl"] = 5

dvd$pcr_repl = "NA"
dvd[grep("DvD[0-9]{1,2}[.][0-9]{1}[.]{0}[0-9]{0}", rownames(dvd)), "pcr_repl"] = 1
dvd[grep("DvD[0-9]{1,2}[.][0-9]{1}[.]{1}1", rownames(dvd)), "pcr_repl"] = 2
dvd[grep("DvD[0-9]{1,2}[.][0-9]{1}[.]{1}2", rownames(dvd)), "pcr_repl"] = 3
dvd[grep("DvD[0-9]{1,2}[.][0-9]{1}[.]{1}3", rownames(dvd)), "pcr_repl"] = 4

# for the control replicates
dvd[grep("DvD[0-9]{1,2}C[.]{0}[0-9]{0}", rownames(dvd)), "pcr_repl"] = 1
dvd[grep("DvD[0-9]{1,2}C[.]{1}1", rownames(dvd)), "pcr_repl"] = 2
dvd[grep("DvD[0-9]{1,2}C[.]{1}2", rownames(dvd)), "pcr_repl"] = 3
dvd[grep("DvD[0-9]{1,2}C[.]{1}3", rownames(dvd)), "pcr_repl"] = 4

# Feeding trials with Asellus
#removing pcr and sample replicate numbers
dva$sample_id = gsub("[.][0-9]{1}$", "", gsub("[.][0-9]{1}$", "", dva$sample_id))

#setting pcr and sample repliate numbers in separate columns
dva$trial_repl = "NA"
dva[grep("C", rownames(dva)), "trial_repl"] = "control"
dva[grep("DvA[0-9]{1,2}[.]1[.]{0,1}[0-9]{0,1}", rownames(dva)), "trial_repl"] = 1
dva[grep("DvA[0-9]{1,2}[.]2[.]{0,1}[0-9]{0,1}", rownames(dva)), "trial_repl"] = 2
dva[grep("DvA[0-9]{1,2}[.]3[.]{0,1}[0-9]{0,1}", rownames(dva)), "trial_repl"] = 3
dva[grep("DvA[0-9]{1,2}[.]4[.]{0,1}[0-9]{0,1}", rownames(dva)), "trial_repl"] = 4
dva[grep("DvA[0-9]{1,2}[.]5[.]{0,1}[0-9]{0,1}", rownames(dva)), "trial_repl"] = 5

dva$pcr_repl = "NA"
dva[grep("DvA[0-9]{1,2}[.][0-9]{1}[.]{0}[0-9]{0}", rownames(dva)), "pcr_repl"] = 1
dva[grep("DvA[0-9]{1,2}[.][0-9]{1}[.]{1}1", rownames(dva)), "pcr_repl"] = 2
dva[grep("DvA[0-9]{1,2}[.][0-9]{1}[.]{1}2", rownames(dva)), "pcr_repl"] = 3
dva[grep("DvA[0-9]{1,2}[.][0-9]{1}[.]{1}3", rownames(dva)), "pcr_repl"] = 4

# for the control replicates
dva[grep("DvA[0-9]{1,2}C[.]{0}[0-9]{0}", rownames(dva)), "pcr_repl"] = 1
dva[grep("DvA[0-9]{1,2}C[.]{1}1", rownames(dva)), "pcr_repl"] = 2
dva[grep("DvA[0-9]{1,2}C[.]{1}2", rownames(dva)), "pcr_repl"] = 3
dva[grep("DvA[0-9]{1,2}C[.]{1}3", rownames(dva)), "pcr_repl"] = 4


# Feeding trials with Crangonyx
#removing pcr and sample replicate numbers
dvc$sample_id = gsub("[.][0-9]{1}$", "", gsub("[.][0-9]{1}$", "", dvc$sample_id))

#setting pcr and sample repliate numbers in separate columns
dvc$trial_repl = "NA"
dvc[grep("C", rownames(dvc)), "trial_repl"] = "control"
dvc[grep("DvG[0-9]{1,2}[.]1[.]{0,1}[0-9]{0,1}", rownames(dvc)), "trial_repl"] = 1
dvc[grep("DvG[0-9]{1,2}[.]2[.]{0,1}[0-9]{0,1}", rownames(dvc)), "trial_repl"] = 2
dvc[grep("DvG[0-9]{1,2}[.]3[.]{0,1}[0-9]{0,1}", rownames(dvc)), "trial_repl"] = 3
dvc[grep("DvG[0-9]{1,2}[.]4[.]{0,1}[0-9]{0,1}", rownames(dvc)), "trial_repl"] = 4
dvc[grep("DvG[0-9]{1,2}[.]5[.]{0,1}[0-9]{0,1}", rownames(dvc)), "trial_repl"] = 5

dvc$pcr_repl = 1
```

We remove all taxa that have sum DNA reads equal to 0 and then we check the remaining taxa as contamination. We can then keep only the main target species of the trials _D.magna_, _A.aquaticus_ and _C.pseudogracilis_.

```{r removing unnecessary taxa}
dvd.info = dvd[,c(colnames(dvd) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]
dva.info = dva[,c(colnames(dva) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]
dvc.info = dvc[,c(colnames(dvc) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]

colSums(dvd[,!c(colnames(dvd) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))])
colSums(dva[,!c(colnames(dva) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))])
colSums(dvc[,!c(colnames(dvc) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))])

dvd = subset(dvd[,!c(colnames(dvd) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))], select=colSums(dvd[,!c(colnames(dvd) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]) > 0)

dva = subset(dva[,!c(colnames(dva) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))], select=colSums(dva[,!c(colnames(dva) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]) > 0)

dvc = subset(dvc[,!c(colnames(dvc) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))], select=colSums(dvc[,!c(colnames(dvc) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]) > 0)

dvd = cbind(dvd, dvd.info)
dva = cbind(dva, dva.info)
dvc = cbind(dvc, dvc.info)
```

 The only conamination we retained is very low (N reads = 7) of Human DNA in the feeding trials with _C.pseudogracilis_ (DvG). We feel confident in removing it.

```{r removing DvG contamination}
colSums(dvc[,!c(colnames(dvc) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))])

dvc = dvc[,!c(colnames(dvc) %in% "Homo_sapiens")]
```

Because we obtained the results from two separate sequencing libraries we need to normalise the DNA reads.

```{r normalised DNA reads}
dvd.info = dvd[,c("sample_id","sample_type","trial_repl","pcr_repl")]
dva.info = dva[,c("sample_id","sample_type","trial_repl","pcr_repl")]
dvc.info = dvc[,c("sample_id","sample_type","trial_repl","pcr_repl")]

dvd.norm = (dvd[,!c(colnames(dvd) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]/sum(feb_clean[,!c(colnames(feb_clean) %in% c("sample_id","sample_type"))]))*1000000

dva.norm = (dva[,!c(colnames(dva) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]/sum(feb_clean[,!c(colnames(feb_clean) %in% c("sample_id","sample_type"))]))*1000000

dvc.norm = (dvc[,!c(colnames(dva) %in% c("sample_id","sample_type","trial_repl","pcr_repl"))]/sum(aug_clean[,!c(colnames(aug_clean) %in% c("sample_id","sample_type"))]))*1000000

dvd.norm = cbind(dvd.norm, dvd.info)
dva.norm = cbind(dva.norm, dva.info)
dvc.norm = cbind(dvc.norm, dvc.info)
```

Once we sorted the frames out we need to sum the sequences based on the pcr replicates. This need to be performed only for DvD and DvA trials

```{r summing pcr replicates}
dvd.trials = as.data.frame(dvd.norm[,!c(colnames(dvd.norm) %in% "pcr_repl")] %>% group_by(sample_id, sample_type, trial_repl) %>% summarise_all(list(sum)))

dva.trials = as.data.frame(dva.norm[,!c(colnames(dva.norm) %in% "pcr_repl")] %>% group_by(sample_id, sample_type, trial_repl) %>% summarise_all(list(sum)))

dvc.trials = dvc.norm[,!c(colnames(dvc.norm) %in% "pcr_repl")]
```


```{r plot for ggplot}
daph.melt = melt(dvd.trials)
asel.melt = melt(dva.trials)
cran.melt = melt(dvc.trials)

ggplot(daph.melt, aes(trial_repl, value)) +
  geom_boxplot(aes(fill=variable)) +
  facet_grid(variable~., scales="free")

ggplot(asel.melt, aes(trial_repl, value)) +
  geom_boxplot(aes(fill=variable)) +
  facet_grid(variable~., scales="free")

ggplot(cran.melt, aes(trial_repl, value)) +
  geom_boxplot(aes(fill=variable)) +
  facet_grid(variable~., scales="free")
```

```{r adding time info}
dvd.trials$times = sub('^\\w{3}', "", dvd.trials$sample_id)
dva.trials$times = sub('^\\w{3}', "", dva.trials$sample_id)
dvc.trials$times = sub("[.]", "", sub('^\\w{3}', "", dvc.trials$sample_id))
```


```{r setting up feeding trials}
dvd.melt = melt(dvd.trials)
dva.melt = melt(dva.trials)
dvc.melt = melt(dvc.trials)

# We split the actual trial from the controls.
dvd.contr = dvd.melt[grep("C", dvd.melt$sample_id),]
dvd.melt = dvd.melt[grep("C", dvd.melt$sample_id, invert=T),]

dva.contr = dva.melt[grep("C", dva.melt$sample_id),]
dva.melt = dva.melt[grep("C", dva.melt$sample_id, invert=T),]

dvc.contr = dvc.melt[grep("C", dvc.melt$sample_id),]
dvc.melt = dvc.melt[grep("C", dvc.melt$sample_id, invert=T),]

dvd.melt$times = factor(dvd.melt$times, levels = c("0","4","8","12","24","36"))
dva.melt$times = factor(dva.melt$times, levels = c("0","4","8","12","24","36"))
dvc.melt$times = factor(dvc.melt$times, levels = c("0","4","8","12","24","36"))
dvc.melt$times[is.na(dvc.melt$times)] <- 12
```

```{r plotting feeding trials}
trials.melt = full_join(dvd.melt, dva.melt)
trials.melt = full_join(trials.melt, dvc.melt)

png("R_images/feeding_trials_boxplot.png", width=5000, height=4000, units="px", res=300)
ggplot(trials.melt[trials.melt$variable != "Dikerogammarus_villosus",], aes(times, value)) +
  geom_boxplot(outlier.shape=NA) +
  geom_jitter(alpha = 0.4) +
  facet_grid(variable~., scale="free") +
  theme_bw() +
  labs(x="Digestion time",y="Normalised DNA reads")
dev.off()
```

# 6. Prey detections in gut contents

One season only across 3 lakes: Wroxham Broad, Grafham Water and Rockland Broad (control site).

```{r lakes and gut contents}
edna = aug_clean[aug_clean$sample_type == "eDNA",]
guts = aug_clean[aug_clean$sample_type == "gut contents",]
```

Remove empty taxa before continuing.

```{r removing empty taxa}
edna.info = edna[,c(colnames(edna) %in% c("sample_id","sample_type"))]
guts.info = guts[,c(colnames(guts) %in% c("sample_id","sample_type"))]

edna = subset(edna[,!c(colnames(edna) %in% c("sample_id","sample_type"))], select=colSums(edna[,!c(colnames(edna) %in% c("sample_id","sample_type"))]) > 0)
guts = subset(guts[,!c(colnames(guts) %in% c("sample_id","sample_type"))], select=colSums(guts[,!c(colnames(guts) %in% c("sample_id","sample_type"))]) > 0)

colSums(edna)
colSums(guts)

rowSums(edna)
rowSums(guts)
```


```{r removing predator based on id}
# Adding predator ID base on DNA ratio greater than 0.5 and 0.9
predator = which(guts[,]/rowSums(guts) >= 0.8, arr.ind=T)
dim(guts)
colnames(guts[rownames(predator),as.vector(predator[,2])])

guts$predator_id = "NA"
guts[rownames(predator),"predator_id"] = gsub("[.]?[0-9]{1,3}$", "", colnames(guts[rownames(predator),as.vector(predator[,2])]))

# Adding missing predators
count(guts$predator_id == "NA")
miss.pred = rownames(guts[guts$predator_id == "NA",])
guts[rownames(guts) %in% miss.pred,]

guts[rownames(guts) %in% miss.pred,!c(colnames(guts) %in% "predator_id")]/rowSums(guts[rownames(guts) %in% miss.pred, !c(colnames(guts) %in% "predator_id")])

guts$predator_id[rownames(guts) == "WB.507"] = "Dikerogammarus_villosus"
guts$predator_id[rownames(guts) == c("RB.303","RB.304")] = "Corophium_multisetosum"
guts$predator_id[c(rownames(guts) %in% c("RB.305","RB.401","RB.503","RB.507","RB.522","RB.524","RB.530","RB.603","RB.607","RB.608","RB.610","RB.611"))] = "Gammarus_zaddachi"
```


```{r merging data with metadata again}
guts = cbind(guts.info, guts)
edna = cbind(edna.info, edna)
```

Adding predator id, site and lake information.
 
```{r removing taxa plus adding metadata}
colnames(guts)

guts$site = gsub("[.]","", gsub("[0-9]{1,2}$", "", rownames(guts)))

guts$sample_id = gsub("[.]","" , guts$sample_id)
rownames(guts) = guts$sample_id

guts$predator_id = gsub("[a-z]{5}$", "", gsub("[a-z]{7,}_","",guts$predator_id, "_"))

guts$lake = NA
guts[grep("WB", guts$site),"lake"] = "WB"
guts[grep("GW", guts$site),"lake"] = "GW"
guts[grep("RB", guts$site),"lake"] = "RB"
```

From the rarefaction curves of both species Richness and Shannon diversity extrapolated from the gut contents analysis it appear that more individuals might need to be collected in order to increase the number of detections from the gut contents.

```{r abundance in guts, include=T, echo=F}
abund.guts = as.data.frame(guts)
abund.guts[,!c(colnames(abund.guts) %in% c("sample_id","sample_type","site","lake","predator_id"))][abund.guts[,!c(colnames(abund.guts) %in% c("sample_id","sample_type","site","lake","predator_id"))] > 0] <- 1
```

We need to remove the predators from gut contents

```{r removing predators from guts}
unique(abund.guts$predator_id)

abund.guts[abund.guts$predator_id == "Dvil","Dikerogammarus_villosus"] = 0
abund.guts[abund.guts$predator_id == "Gzad","Gammarus_zaddachi"] = 0
abund.guts[abund.guts$predator_id == "Gtig","Gammarus_tigrinus"] = 0
abund.guts[abund.guts$predator_id == "Cmultise","Corophium_multisetosum"] = 0

abund.guts = abund.guts[,!c(colnames(abund.guts) %in% "Dikerogammarus_villosus")]
```

```{r rarefaction curves}
raref.guts = as.data.frame(abund.guts[grep("Dvil|Gzad", abund.guts$predator_id),!c(colnames(abund.guts) %in% c("sample_id","sample_type","site"))] %>% group_by(lake, predator_id) %>% summarise_all(list(sum)))

rownames(raref.guts) = paste(raref.guts$predator_id,raref.guts$lake)
colnames(raref.guts)

raref.guts = t(raref.guts[,!c(colnames(raref.guts) %in% c("lake","predator_id"))])

raref.guts_inext = iNEXT(raref.guts, q=c(0,1), datatype = "abundance", se=T, endpoint=150)

strip_lab=c("0"="Species Richness","1"="Shannon Diversity")

png("R_images/guts_rarefactions.png", width=4000, height=4000, units="px", res=300)
ggiNEXT(raref.guts_inext) +
  labs(y="", x="Number detections") +
  theme(legend.position="bottom") +
  facet_grid(order~., labeller=as_labeller(strip_lab), scales="free")
dev.off()
```

PERMANOVA analysis on the gut contents show a significant influence from all variables (season: p-value=9.999e-05; county: p-value=9.999e-05; both: p-value=2e-04). It worths mentioning though that the residual values for these samples are all high; so we can assume that the influence of the low number of samples positive for prey taxa could have skewed the results.

```{r permanova guts reads, include=T, echo=F}
adonis.guts = as.data.frame(abund.guts[grep("Dvil|Gzad", abund.guts$predator_id),])

fact_site = matrix(c(adonis.guts$site),ncol=1)
fact_lake = matrix(c(adonis.guts$lake),ncol=1)
fact_pred = matrix(c(adonis.guts$predator_id),ncol=1)

adonis.guts = adonis.guts[,!c(colnames(adonis.guts) %in% c("sample_id","sample_type","site","lake","predator_id"))]

dist.guts = vegdist(adonis.guts, method="euclidean")
adonis(dist.guts~fact_site*fact_pred*fact_lake, permutations = 9999, method="euclidean")
```

```{r metamds guts, include=F, echo=F}
meta.guts = metaMDS(dist.guts, zerodist=ignore, trymax=200)
```

```{r permanova guts plot, include=T, echo=F}
png("R_images/NMDS_gut_contents.png", width=4000, height=4000, units="px", res=300)
plot(meta.guts,type="n", xlim=c(-2,2), ylim=c(-1.5,1.5))
points(meta.guts, select=which(fact_lake[,1]=="WB"), col="black",pch=19)
points(meta.guts, select=which(fact_lake[,1]=="GW"), col="blue",pch=15)
points(meta.guts, select=which(fact_lake[,1]=="RB"), col="red",pch=17)
ordispider(meta.guts,group=fact_lake[,1], col=c("blue","orange","black"),label=F)
ordiellipse(meta.guts,group=fact_pred[,1],kind="ehull", lty=c(1,5),label=T,lwd=2)
text(-0.5,-1.1,label=paste("Stress:", round(meta.guts$stress, 6)))
dev.off()
```

# 7. Bipartite network

```{r setting up the bipartite}
abund.guts = abund.guts[,!c(colnames(abund.guts) %in% "Diptera")]

web.guts = as.data.frame(abund.guts[,!c(colnames(abund.guts) %in% c("sample_id","sample_type","site"))] %>% group_by(lake, predator_id) %>% summarise_all(list(sum)))

rownames(web.guts) = paste(web.guts$predator_id, web.guts$lake)

web.guts = web.guts[,!c(colnames(web.guts) %in% c("predator_id","lake"))]

web.guts = web.guts[!c(rownames(web.guts) %in% c("Gtig GW","Cmultise RB")),]

web.guts = t(web.guts)

rownames(web.guts) = gsub("[a-z]{3,}_",".",rownames(web.guts))
```

```{r plot order}
seq.high=c("Dvil GW","Dvil WB","Gzad RB")

seq.low=c("C.luridus","C.laricomalis","G.tigrinus","B.vejdovskyanum","C.brevilabris","Cricotopus","Cyclopidae","Daphnia","D.polymorpha","G.zaddachi","P.pediculus","A.sieboldi","C.diastrophus","C.sp._BOLD:AAI4299","C.multisetosum","C.floridanus","C.pseudogracilis","E.dilatata","Hydra","H.oligactis","L.phaeopa","R.auricularia","S.sthenum","S.lacustris")

col.network=c("yellow","yellow","red","tan4","cyan3","yellow","cyan3","cyan3","darkorchid3","red","cyan3","darkgreen","tan4","yellow","red","red","red","darkgreen","darkcyan","darkcyan","khaki3","darkorchid3","grey48","tan4")
```

```{r legend order}
name.multi=c("Cladocera","Amphipoda","Diptera","Hydrozoa","Mollusca","Oligochaeta","Platyhelminthes","Rotifera","Trichoptera")
col.multi=c("cyan3","red","yellow","darkcyan","darkorchid3","tan4","grey48","darkgreen","khaki3")
```

After having set up the orders of taxa in top and bottom levels with their corresponding colours, the bipartite plot and the legend can be generated. The legend was generated separately due to limits in the plot dimensions, which if reduced would have made either the plot, or the legend non readable. The legend and the plot were merged externally to R.

```{r plotting the networks}
# Producing the bipartite network
png("R_images/bipartite_network.png", width=4000, height=2500, units="px", res=300)
plotweb(sortweb(web.guts, sort.order="seq", sequence=list(seq.higher=seq.high, seq.lower=seq.low)), method = "normal", y.width.low=0.1, y.width.high=0.08,	y.lim=c(0,3), labsize=2, low.lab.dis=0.05, high.lab.dis=0.1, low.spacing=0.012, high.spacing = 0.11, text.rot=90, col.high="black", col.low=col.network, high.y=2.5, low.y=1.5)
# Producing the legend
legend("bottom", legend=name.multi, fill=col.multi, border="black", bty="n", ncol=3, xpd=T, cex=1.7)
dev.off()
```


# 8. Community analysis

```{r edna frame}
edna$site = gsub("^e", "", rownames(edna))
edna$site

edna$lake = NA
edna[grep("GW", edna$site),"lake"] = "GW"
edna[grep("WB", edna$site),"lake"] = "WB"
edna[grep("RB", edna$site),"lake"] = "RB"
edna$lake
```

```{r abundance edna}
abund.edna = as.data.frame(edna)
colnames(abund.edna)
abund.edna[,!c(colnames(abund.edna) %in% c("sample_id","sample_type","site","lake"))][abund.edna[,!c(colnames(abund.edna) %in% c("sample_id","sample_type","site","lake"))] > 0] <- 1
```

Rerefaction curves eDNA samples.

```{r edna rarefaction curves}
raref.edna = as.data.frame(abund.edna[,!c(colnames(abund.edna) %in% c("sample_id","sample_type"))] %>% group_by(site,lake) %>% summarise_all(list(sum)))

rownames(raref.edna) = raref.edna$site
colnames(raref.edna)

raref.edna = t(raref.edna[,!c(colnames(raref.edna) %in% c("lake","site"))])

raref.edna_inext = iNEXT(raref.edna, q=c(0,1), datatype = "abundance", se=T, endpoint=50)

strip_lab=c("0"="Species Richness","1"="Shannon Diversity")

png("R_images/edna_rarefactions.png", width=4000, height=4000, units="px", res=300)
ggiNEXT(raref.edna_inext) +
  labs(y="", x="Number detections") +
  theme(legend.position="bottom") +
  facet_grid(order~., labeller=as_labeller(strip_lab), scales="free")
dev.off()
```

PERMANOVA analysis on the eDNA samples

```{r permanova guts reads, include=T, echo=F}
adonis.edna = as.data.frame(abund.edna)

fact_esite = matrix(c(adonis.edna$site),ncol=1)
fact_elake = matrix(c(adonis.edna$lake),ncol=1)

adonis.edna = adonis.edna[,!c(colnames(adonis.edna) %in% c("sample_id","sample_type","site","lake"))]

dist.edna = vegdist(adonis.edna)
adonis(dist.edna~fact_esite*fact_elake, permutations = 9999, method="jaccard")
```

```{r metamds guts, include=F, echo=F}
meta.edna = metaMDS(dist.edna, distance="jaccard", trymax=200)
```

```{r permanova guts plot, include=T, echo=F}
png("R_images/NMDS_edna_samples.png", width=4000, height=4000, units="px", res=300)
plot(meta.edna,type="n", xlim=c(-0.5,0.5), ylim=c(-0.5,0.5))
points(meta.edna, select=which(fact_elake[,1]=="WB"), col="black",pch=19)
points(meta.edna, select=which(fact_elake[,1]=="GW"), col="blue",pch=15)
points(meta.edna, select=which(fact_elake[,1]=="RB"), col="red",pch=17)
ordispider(meta.edna,group=fact_elake[,1], col=c("blue","orange","black"),label=T)
ordiellipse(meta.edna,group=fact_elake[,1],kind="ehull", lty=c(1,3,5),label=F,lwd=2)
text(-0.5,-1.1,label=paste("Stress:", round(meta.edna$stress, 6)))
dev.off()
```

# 9. Microscopy community

Microscopy data from the community collected with the kick-samples are available in 6 separate files belonging to each lake from either sampling season.

```{r input files microscopy}
gw.may = read.csv("data/microscopy/gw_may_microscopy.csv", sep="\t")
wb.may = read.csv("data/microscopy/wb_may_microscopy.csv", sep="\t")
rb.may = read.csv("data/microscopy/rb_may_microscopy.csv", sep="\t")
```

Before analysis the data we remove the high rank taxa.

```{r adding the binomial}
data_structuring <- function(df){
  df = as.data.frame(df)
  df = df[,!c(colnames(df) %in% c("ORDER","FAMILY"))]
  df = spread(df, TAXA, NUMBERS, fill=0)
  return(df)
}

gw.may.kick = data_structuring(gw.may)
wb.may.kick = data_structuring(wb.may)
rb.may.kick = data_structuring(rb.may)

# Adding variables

gw.may.kick$sample_type = "kick"
wb.may.kick$sample_type = "kick"
rb.may.kick$sample_type = "kick"

gw.may.kick$lake = "GW"
wb.may.kick$lake = "WB"
rb.may.kick$lake = "RB"

gw.kick = as.data.frame(gw.may.kick)
wb.kick = as.data.frame(wb.may.kick)
rb.kick = as.data.frame(rb.may.kick)
```

```{r cleaning datasets}
# Removing unnecessary columns
gw.kick = gw.kick[,!c(colnames(gw.kick) %in% c("TUBE.ID","DATE"))]
wb.kick = wb.kick[,!c(colnames(wb.kick) %in% c("TUBE.ID","DATE"))]
rb.kick = rb.kick[,!c(colnames(rb.kick) %in% c("TUBE.ID","DATE"))]

gw.kick$sample_id = paste0(gw.kick$LOCATION, gw.kick$REPLICATE)
wb.kick$sample_id = paste0(wb.kick$LOCATION, wb.kick$REPLICATE)
rb.kick$sample_id = paste0(rb.kick$LOCATION, rb.kick$REPLICATE)
```

```{r joining frames}
abund.kick = full_join(gw.kick, wb.kick)
abund.kick = full_join(abund.kick, rb.kick)
abund.kick[is.na(abund.kick)] <- 0

colnames(abund.kick)[colnames(abund.kick) == "LOCATION"] <- "site"

abund.kick$site =  abund.kick$sample_id
```

```{r rarefaction curves on kick-samples}
# removing high order taxa
abund.kick = abund.kick[,!c(colnames(abund.kick) %in% c("Trichoptera","Diptera","Oligochaeta","Ostracoda","Araneae","Coleoptera","Mollusca","Zygoptera","Plecoptera","Nematoda"))]

# removing extra variables
raref.kick = abund.kick[,!c(colnames(abund.kick) %in% c("sample_id","REPLICATE","sample_type"))]

# list of families in data set to be removed
kick.fam = colnames(raref.kick[,grepl("idae", colnames(raref.kick)) == T])
raref.sp.kick = raref.kick[,!c(colnames(raref.kick) %in% kick.fam)]

# summarising data set by lake and season
raref.sp.kick = as.data.frame(raref.sp.kick %>% group_by(lake,site) %>% summarise_all(list(sum)))

rownames(raref.sp.kick) = raref.sp.kick$site
raref.sp.kick[is.na(raref.sp.kick)] <- 0
raref.sp.kick = t(raref.sp.kick[,!c(colnames(raref.sp.kick) %in% c("site","lake"))])

inext.kick = iNEXT(raref.sp.kick, q=c(0,1), datatype = "abundance", se=T, endpoint=1400)

strip_labels=c("0"="Species Richness","1"="Shannon Diversity")
png("R_images/kick-samples_rarefactions.png", width = 3000, height = 3000, units = "px", res=300)
ggiNEXT(inext.kick) +
  labs(y = "", x="Number of individuals") +
  theme(legend.position="right") +
  facet_grid(order~., labeller=as_labeller(strip_labels), scales="free") +
  ggtitle("Rarefaction curves of communities in kick samples")
dev.off()
```

Calculating PERMANOVA of kick-samples

```{r PERMANOVA of kick-samples}
occup.kick = as.data.frame(abund.kick[,!c(colnames(abund.kick) %in% c("REPLICATE"))] %>% group_by(site,sample_id,sample_type,lake) %>% summarise_all(list(sum)))

rownames(occup.kick) = occup.kick$site

occup.kick[,!c(colnames(occup.kick) %in% c("sample_id","site","sample_type","lake"))][occup.kick[,!c(colnames(occup.kick) %in% c("sample_id","site","sample_type","lake"))] > 0] <- 1

adonis.kick = as.data.frame(occup.kick)

fact_site.kick = matrix(c(adonis.kick$site), ncol=1)
fact_lake.kick = matrix(c(adonis.kick$lake), ncol=1)

adonis.kick = adonis.kick[,!c(colnames(adonis.kick) %in% c("sample_id","site","sample_type","lake"))]

dist.kick = vegdist(as.matrix(adonis.kick))

adonis(dist.kick~fact_lake.kick*fact_site.kick, permutations = 9999, method="jaccard")
```

```{r}
meta.kick = metaMDS(dist.kick, trymax=200, distance="jaccard")
```

```{r NMDS plot of kick-samples}
# full symbols for May, hollow symbols for October
png("R_images/NMDS_kick_samples.png", width = 3000, height = 3000, units = "px", res = 300)
plot(meta.kick, type="n", main="kick-samples NMDS", xlim=c(-0.5,0.5), ylim=c(-0.5,0.5))
points(meta.kick, select=which(fact_lake.kick[,1]=="WB"),col="black",pch=20)
points(meta.kick, select=which(fact_lake.kick[,1]=="GW"),col="blue",pch=15)
points(meta.kick, select=which(fact_lake.kick[,1]=="RB"),col="red",pch=17)
ordispider(meta.kick, group=fact_lake.kick[,1],col=c("blue","orange","green"),label=T)
ordiellipse(meta.kick, group=fact_lake.kick[,1],kind="ehull",lty=c(1,3,5),label=F,lwd=1)
text(0.2,-0.4, label=paste0("Stress: ", round(meta.kick$stress,6)))
dev.off()
```

Calculating species richness for kick-samples and eDNA samples

```{r diversity indices kick-samples}
diversity.kick = occup.kick
diversity.kick[,!c(colnames(diversity.kick) %in% c("sample_id","site","sample_type","lake"))][diversity.kick[,!c(colnames(diversity.kick) %in% c("sample_id","site","sample_type","lake"))] > 0] <- 1

diversity.kick[grep("GW|WB", rownames(diversity.kick)), grep("Dikerogammarus", colnames(diversity.kick))] <- 0
diversity.kick[grep("RB", rownames(diversity.kick)), grep("Gammarus_zaddachi|Gammarus_sp.", colnames(diversity.kick))] <- 0

richness.kick = rowSums(diversity.kick[,!c(colnames(diversity.kick) %in% c("sample_id","site","sample_type","lake"))])

richness.kick = cbind(richness.kick, occup.kick[,c(colnames(occup.kick) %in% c("sample_id","sample_type","site","lake"))])
colnames(richness.kick)[colnames(richness.kick) == "richness.kick"] <- "richness"
```

```{r diversity indices eDNA}
diversity.edna = abund.edna
diversity.edna[,!c(colnames(diversity.edna) %in% c("sample_id","site","sample_type","lake"))][diversity.edna[,!c(colnames(diversity.edna) %in% c("sample_id","site","sample_type","lake"))] > 0] <- 1

diversity.edna[grep("GW|WB", rownames(diversity.edna)), grep("Dikerogammarus", colnames(diversity.edna))] <- 0
diversity.edna[grep("RB", rownames(diversity.edna)), grep("Gammarus_zaddachi|Gammarus_sp.", colnames(diversity.edna))] <- 0

richness.edna = rowSums(diversity.edna[,!c(colnames(diversity.edna) %in% c("sample_id","site","sample_type","lake"))])

richness.edna = cbind(richness.edna, abund.edna[,c(colnames(abund.edna) %in% c("sample_id","sample_type","site","lake"))])
colnames(richness.edna)[colnames(richness.edna) == "richness.edna"] <- "richness"

richness.full = full_join(richness.edna, richness.kick)
```

```{r making metadata consistent}
richness.full[richness.full$sample_type == "kick", "site"] = richness.full[richness.full$sample_type == "kick", "sample_id"]
```

```{r richness boxplots}
png("R_images/species_richness_community_boxplot.png", width=3000, height=3000, units="px", res=300)
richness.full %>% mutate(site = factor(lake, levels=c("GW","WB","RB"))) %>% 
  ggplot(aes(site, richness)) +
  geom_boxplot(aes(fill=sample_type), outlier.shape = NA, alpha=0.9) +
  geom_point(position=position_jitterdodge(dodge.width = 0.85), aes(fill=sample_type), color="black", pch=21, size=2, alpha=0.5) +
  facet_grid(.~site, scales = "free_x") +
  theme_bw() + labs(y="Species richness", x="") +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))
dev.off()

shannon.kick = diversity(diversity.kick[,!c(colnames(diversity.kick) %in% c("site","sample_id","sample_type","time","lake"))], index="shannon")
shannon.kick = cbind(shannon.kick, diversity.kick[,c(colnames(diversity.kick) %in% c("site","sample_id","sample_type","time","lake"))])
colnames(shannon.kick)[colnames(shannon.kick) == "shannon.kick"] <- "diversity"

png("R_images/shannon_kick-samples_boxplot.png", width=3000, height=3000, units="px", res=300)
shannon.kick %>% mutate(site = factor(site, levels=c("GW","WB","RB"))) %>% 
  ggplot(aes(site, diversity)) +
  geom_boxplot(aes(fill=sample_type), outlier.shape = NA, alpha=0.9) +
  geom_point(position=position_jitterdodge(dodge.width = 0.85), aes(fill=sample_type), color="black", pch=21, size=2, alpha=0.5) +
#  facet_grid(.~sample_type, scales = "free_x") +
  theme_bw() + labs(y="Shannon diversity", x="") +
  theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))
dev.off()
```


```{r richness in eDNA samples}
# GW May
gw_ednarich = richness.edna[grep("GW.", rownames(richness.edna)),!c(colnames(richness.edna) %in% c("sample_id","sample_type","site","lake"))]
mean(gw_ednarich)
sd(gw_ednarich)/sqrt(length(gw_ednarich))

# WB May
wb_ednarich = richness.edna[grep("WB.", rownames(richness.edna)),!c(colnames(richness.edna) %in% c("sample_id","sample_type","site","lake"))]
mean(wb_ednarich)
sd(wb_ednarich)/sqrt(length(wb_ednarich))

# RB May
rb_ednarich = richness.edna[grep("RB.", rownames(richness.edna)),!c(colnames(richness.edna) %in% c("sample_id","sample_type","site","lake"))]
mean(rb_ednarich)
sd(rb_ednarich)/sqrt(length(rb_ednarich))
```


```{r richness in kick samples}
# GW May
gw_kickrich = richness.kick[grep("GW.", rownames(richness.kick)),!c(colnames(richness.kick) %in% c("sample_id","sample_type","site","lake"))]
mean(gw_kickrich)
sd(gw_kickrich)/sqrt(length(gw_kickrich))

# WB May
wb_kickrich = richness.kick[grep("WB.", rownames(richness.kick)),!c(colnames(richness.kick) %in% c("sample_id","sample_type","site","lake"))]
mean(wb_kickrich)
sd(wb_kickrich)/sqrt(length(wb_kickrich))

# RB May
rb_kickrich = richness.kick[grep("RB.", rownames(richness.kick)),!c(colnames(richness.kick) %in% c("sample_id","sample_type","site","lake"))]
mean(rb_kickrich)
sd(rb_kickrich)/sqrt(length(rb_kickrich))
```

# 10. Community vs eDNA

```{r PERMANOVA kick vs eDNA}
abund.edna.kick = full_join(abund.edna, occup.kick)

fact_edki.type = matrix(c(abund.edna.kick$sample_type), ncol=1)
fact_edki.site = matrix(c(abund.edna.kick$site), ncol=1)
fact_edki.lake = matrix(c(abund.edna.kick$lake), ncol=1)

colnames(abund.edna.kick)

adonis.edna.kick = as.data.frame(abund.edna.kick[,!c(colnames(abund.edna.kick) %in% c("sample_type","sample_id","lake","site"))])

dist.edna.kick = vegdist(adonis.edna.kick, method="jaccard")

adonis(dist.edna.kick~fact_edki.lake*fact_edki.site*fact_edki.type, permutations=9999, method="jaccard")
```

```{r NMDS kick and eDNA}
meta.edki = metaMDS(dist.edna.kick, zerodist=ignore, trymax=200)
```

```{r NMDS plot kick vs eDNA}
png("R_images/NMDS_kick_edna.png", width=4000, height=4000, units="px", res=300)
plot(meta.edki,type="n", xlim=c(-0.5,0.5), ylim=c(-0.5,0.5))
points(meta.edki, select=which(fact_edki.lake[,1]=="WB"), col="black",pch=19)
points(meta.edki, select=which(fact_edki.lake[,1]=="GW"), col="blue",pch=15)
points(meta.edki, select=which(fact_edki.lake[,1]=="RB"), col="red",pch=17)
ordispider(meta.edki,group=fact_edki.lake[,1], col=c("blue","orange","black"),label=T)
ordiellipse(meta.edki,group=fact_edki.type[,1],kind="ehull", lty=c(1,5),label=F,lwd=2)
text(-0.5,-1.1,label=paste("Stress:", round(meta.edki$stress, 6)))
dev.off()
```

Then by adding the gut contents to the analysis

```{r}
abund.edna.kick.guts = full_join(abund.edna.kick, abund.guts)

fact_full.type = matrix(c(abund.edna.kick.guts$sample_type), ncol=1)
fact_full.site = matrix(c(abund.edna.kick.guts$site), ncol=1)
fact_full.lake = matrix(c(abund.edna.kick.guts$lake), ncol=1)

adonis.edna.kick.guts = as.data.frame(abund.edna.kick.guts[,!c(colnames(abund.edna.kick.guts) %in% c("sample_type","sample_id","lake","site", "predator_id"))])

dist.full = vegdist(adonis.edna.kick.guts, method="euclidean")

adonis(dist.full~fact_full.lake*fact_full.site*fact_full.type, permutations=9999, method="euclidean")
```

```{r}
meta.full = metaMDS(dist.full, zerodist=ignore, trymax=200)
```

```{r}
png("R_images/NMDS_kick_eDNA_guts.png", width=4000, height=4000, units="px", res=300)
plot(meta.full,type="n")#, xlim=c(-0.5,0.5), ylim=c(-0.5,0.5))
points(meta.full, select=which(fact_full.lake[,1]=="WB"), col="black",pch=19)
points(meta.full, select=which(fact_full.lake[,1]=="GW"), col="blue",pch=15)
points(meta.full, select=which(fact_full.lake[,1]=="RB"), col="red",pch=17)
ordispider(meta.full,group=fact_full.lake[,1], col=c("blue","orange","black"),label=T)
ordiellipse(meta.full,group=fact_full.type[,1],kind="ehull", lty=c(1,3,5),label=T,lwd=2)
text(-0.5,-1.1,label=paste("Stress:", round(meta.full$stress, 6)))
dev.off()
```

# 11. Summary table

```{r summary table}
guts.summary = as.data.frame(abund.guts[,!c(colnames(abund.guts) %in% c("sample_id"))] %>% group_by(sample_type,lake,predator_id,site) %>% summarise_all(funs(sum)))
edna.summary = as.data.frame(abund.edna[,!c(colnames(abund.edna) %in% c("sample_id"))] %>% group_by(sample_type,lake,site) %>% summarise_all(funs(sum)))
kick.summary = as.data.frame(abund.kick[,!c(colnames(abund.kick) %in% c("sample_id","REPLICATE"))] %>% group_by(sample_type,lake,site) %>% summarise_all(funs(sum)))

full.data = full_join(guts.summary, edna.summary)
full.data = full_join(full.data, kick.summary)

# Splitting the metadata from data, so that we can remove the NAs into the data set.
full.info = full.data[,c(colnames(full.data) %in% c("sample_type","lake","predator_id","site"))]
full.data = full.data[,!c(colnames(full.data) %in% c("sample_type","lake","predator_id","site"))]
full.data[is.na(full.data)] <- 0

# Binding the frame together using specific order of the metadata
final.data = bind_cols(full.info, full.data)
colnames(final.data) = gsub("_", " ", colnames(final.data))

# Transposing and saving the data frame, we will rearrange the order of the metadata later
final.data = t(final.data)

#Saving final output
write.csv(final.data, "R_outputs/Data_summary_table.csv")
```

# Log files

```{r log session info}
sink("log_session_analysis.txt")
sessionInfo()
citation()
sink()
```

```{r packages citation}
sink("packages_citations.txt")
for (p in pack.list){
 print(citation(p))
}
sink()
```

```{r end}
print("end")
```